Library/Homebrew/brew.sh

```diff
@@ -17,12 +18,19 @@ esac
 # same file under the native architecture
 # These variables are set from the user environment.
 # shellcheck disable=SC2154
-if [[ "${HOMEBREW_CHANGE_ARCH_TO_ARM}" == "1" ]] &&
-   [[ "${HOMEBREW_MACOS}" == "1" ]] &&
-   [[ "$(sysctl -n hw.optional.arm64 2>/dev/null)" == "1" ]] &&
-   [[ "$(sysctl -n sysctl.proc_translated 2>/dev/null)" == "1" ]]
+if [[ "${HOMEBREW_MACOS}" == "1" ]] &&
+   [[ "$(sysctl -n hw.optional.arm64 2>/dev/null)" == "1" ]]
 then
-  exec arch -arm64e "${HOMEBREW_BREW_FILE}" "$@"
+  # used in vendor-install.sh
+  # shellcheck disable=SC2034
+  HOMEBREW_PHYSICAL_PROCESSOR="arm64"
+  HOMEBREW_ROSETTA="$(sysctl -n sysctl.proc_translated)"
```

Comment:
Could potentially be used in `Library/Homebrew/extend/os/mac/hardware/cpu.rb` but not sure if it's worth the effort 🤷🏻

Reply:
Yea, I thought about that, but decided doing
```ruby
ENV["HOMEBREW_ROSETTA"].present? && ENV["HOMEBREW_ROSETTA"] == "1"
```
was less nice.